x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_style_property.tsx

```diff
@@ -242,38 +286,38 @@ export class DynamicStyleProperty extends AbstractStyleProperty {
     };
   }
 
-  pluckCategoricalStyleMetaFromFieldMetaData(fieldMetaData) {
-    if (!this.isCategorical()) {
+  _pluckCategoricalStyleMetaFromFieldMetaData(styleMetaData: StyleMetaData) {
+    if (!this.isCategorical() || !this._field) {
       return null;
     }
 
-    const rootFieldName = this._field.getRootName();
-    if (!fieldMetaData[rootFieldName] || !fieldMetaData[rootFieldName].buckets) {
+    const fieldMeta = styleMetaData[this._field.getRootName()];
+    if (!fieldMeta || !('buckets' in fieldMeta)) {
       return null;
     }
 
-    const ordered = fieldMetaData[rootFieldName].buckets.map((bucket) => {
-      return {
-        key: bucket.key,
-        count: bucket.doc_count,
-      };
-    });
     return {
-      categories: ordered,
+      categories: fieldMeta.buckets.map((bucket) => {
+        return {
+          key: bucket.key,
+          count: bucket.doc_count,
+        };
+      }),
     };
   }
 
-  formatField(value) {
+  formatField(value: string | undefined): string {
     if (this.getField()) {
-      const fieldName = this.getField().getName();
+      const fieldName = this.getFieldName();
       const fieldFormatter = this._getFieldFormatter(fieldName);
-      return fieldFormatter ? fieldFormatter(value) : value;
+      return fieldFormatter ? fieldFormatter(value) : super.formatField(value);
     } else {
-      return value;
+      return super.formatField(value);
     }
   }
 
-  getNumericalMbFeatureStateValue(value) {
+  getNumericalMbFeatureStateValue(value: unknown) {
```

Comment:
in hindsight, this shouldn't really be a method at all, but just a util in `vector_style.js`